src/user_input.rs

```diff
@@ -260,30 +269,78 @@ impl From<InputButton> for InputMode {
 /// Please contact the maintainers if you need support for another type!
 #[non_exhaustive]
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub enum InputButton {
+pub enum InputKind {

```

Comment:
I renamed this, but would you rather keep it `InputButton` in favor of the "virtual button" mentality?

Reply:
I'd like to be able to map a gamepad axis to an action axis as well (not _button_). `InputKind` seems like a decent name if it were going to include both.

Reply:
But it _might_ make more sense to have separate `InputButton` and `InputAxis` enums, especially if we want to be able to add non-gamepad axes in the future (like a mouse-wheel axis, or a touch-drag axis, etc.).

Reply:
> InputKind seems like a decent name if it were going to include both buttons and axes in a single enum.

It seemed more intuitive to me, and I was thinking that maybe it would prevent people from being turned off from investigating the type if they were looking for the Axis inputs.

> But it might make more sense to have separate InputButton and InputAxis enums, especially if we want to be able to add non-gamepad axes in the future (like a mouse-wheel axis, or a touch-drag axis, etc.).

I _think_ that it's best to keep all that in one enum, because the rest of the library is kind of centered around it, and splitting it out would make things more difficult functionally. But I could see adding more variants to it as we expand the kinds of inputs that we accept.